src/infrastructure/validation/logger/ConsoleLogger.java

```diff
@@ -0,0 +1,219 @@
+	
+	/** ANSI escape codes for using color in output */
+	
+	// reset to return to using console default
+	public static final String ANSI_RESET = "\u001B[0m";
+	
+	// ERROR log level with color RED
+	public static final String ANSI_RED = "\u001B[31m";
+	
+	// WARNING log level with color YELLOW
+	public static final String ANSI_YELLOW = "\u001B[33m";
+	
+	// SUCCESS log level with color GREEN
+	public static final String ANSI_GREEN = "\u001B[32m";
+	
+	/** boolean that enables/disables ERROR logs, 
+	 *  defaults to false (no ERROR logs are logged)
+	 */
+	private boolean enableErrorLog = false;
+	
+	/** boolean that enables/disables WARNING logs, 
+	 *  defaults to false (no WARNING logs are logged)
+	 */
+	private boolean enableWarningLog = false;
+
+	/** boolean that enables/disables SUCCESS logs, 
+	 *  defaults to false (no SUCCESS logs are logged)
+	 */
+	private boolean enableSuccessLog = false;
+
+	/** boolean that enables/disables INFO logs, 
+	 *  defaults to false (no INFO logs are logged)
+	 */
+	private boolean enableInfoLog = false;
+
+	/**
+	 *  boolean to enable/disable testMode while logging
+	 *  If in testMode, logs won't have module or timestamp information attached with it. 
+	 */
+	private boolean inTestMode = false;
+	
+	/**
+	 *  Creates an object that logs to the console, if enabled and available
+	 */
+	protected ConsoleLogger(List<LogLevel> enabledLogLevelsList, boolean enableTestMode) {
+		
+		timeStampFormat = DateTimeFormatter.ofPattern("yyyy:MM:dd HH:mm:ss");
+		
+		console = System.console();
+		
+		// set testMode to enabled/disabled
+		// variable enableTestMode is a boolean which is a primitive data type and,
+		// has a default value of false if unassigned
+		inTestMode = enableTestMode;
+		
+		if(null == console) {
+			
+			// no console to print found
+			// disable all the log levels i.e. have no messages logged
+			enableErrorLog = false;
+			enableWarningLog = false;
+			enableSuccessLog = false;
+			enableInfoLog = false;
+		
+		} else if (null == enabledLogLevelsList) {
+		
+			// no info specified
+			// use default value
+			enableErrorLog = false;
+			enableWarningLog = false;
+			enableSuccessLog = false;
+			enableInfoLog = false;
+		
+		} else {
+			
+			// check the List to see which LogLevel enum are sent
+			// for every member of the List, set the corresponding boolean to true i.e. enabled
+			if (enabledLogLevelsList.contains(LogLevel.ERROR)) {
+				enableErrorLog = true;
+			}
+			if (enabledLogLevelsList.contains(LogLevel.WARNING)) {
+				enableWarningLog = true;
+			}
+			if (enabledLogLevelsList.contains(LogLevel.SUCCESS)) {
+				enableSuccessLog = true;
+			}
+			if (enabledLogLevelsList.contains(LogLevel.INFO)) {
+				enableInfoLog = true;
+			}
+			
+		}
+		
+	}
+
+	@Override
+	synchronized public void log(ModuleID moduleIdentifier, LogLevel level, String message) {
+		
+		LocalDateTime now = LocalDateTime.now();
+		String formatDateTime = now.format(timeStampFormat);
```

Comment:
`format` throws `DateTimeException` if an error occurs during printing.
Our program may crash if this exception remains uncaught.

Please use a try-catch block

Reply:
Cool
added it at everywhere I use `format`
I'm not entirely sure what could be done in this case, so I have just went with not having timestamps if this exception occurs
new PR will address these